[TECHNICAL-SUPPORT] LPS-58022 User not found if you search for it in Users and Organizations on the Browse tab

lipusz opened the pull request:
https://issues.liferay.com/browse/LPS-58022

Hello Pei-Jung,

Nice "to meet you". ;-) Jorge suggested me to forward this to you as he's high load nowadays.

Note from István @istvansajtos:
>In Users and Organizations app users are not displayed/found on Browse tab while they are visible on All Users tab.

>I removed unnecessary permission constraints to eliminate the issue and make the behaviour on Browse tab consistent with All Users tab.

>Thanks for checking!

The "regression" was introduced as a side-effect of LPS-25718 (@3b43947ce1c2990b818412198346acdd28fd633f) and LPS-26195 (@d2f5aaa07a17805aa3a8995b80a6e4b72a15eb63).

Note: the search box is not present on sub-levels of organizations in 6.1.x.

Let us know if you any concern.

Thanks,
Tibor

liferay-continuous-integration commented:
<h1>All tests PASSED!</h1><p>Build Time: 1 hour 4 minutes 6 seconds</p><h4>Job Results:</h4><p>16 Jobs Passed.<br />0 Jobs Failed.</p><h1>See the following links for more information.</h1><ul><li><h4><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest(master)/1127/">test-portal-acceptance-pullrequest(master)</a></h4></li><li><h4><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6520/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat-mysql/portal-smoke</a></h4></li><li><h4><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6576/">test-portal-acceptance-pullrequest-batch(master)/integration-db2</a></h4></li><li><h4><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6568/">test-portal-acceptance-pullrequest-batch(master)/integration-mysql</a></h4></li><li><h4><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6613/">test-portal-acceptance-pullrequest-batch(master)/integration-oracle</a></h4></li><li><h4><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6635/">test-portal-acceptance-pullrequest-batch(master)/integration-postgresql</a></h4></li><li><h4><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6558/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-db2</a></h4></li><li><h4><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6519/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql</a></h4></li><li><h4><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6626/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-oracle</a></h4></li><li><h4><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6521/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-postgresql</a></h4></li><li><h4><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6577/">test-portal-acceptance-pullrequest-batch(master)/modules-unit</a></h4></li><li><h4><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6569/">test-portal-acceptance-pullrequest-batch(master)/plugins</a></h4></li><li><h4><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6614/">test-portal-acceptance-pullrequest-batch(master)/portal-web</a></h4></li><li><h4><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6636/">test-portal-acceptance-pullrequest-batch(master)/service-builder</a></h4></li><li><h4><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6559/">test-portal-acceptance-pullrequest-batch(master)/tck</a></h4></li><li><h4><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/6520/">test-portal-acceptance-pullrequest-batch(master)/unit</a></h4></li><li><h4><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-source(master)/1089/">test-portal-acceptance-pullrequest-source(master)</a></h4></li></ul>

pei-jung commented:
Hi Tibor,

Nice to meet you on github too! :P

We can't remove those permission constraints because those are needed to hide the users panel from user who do not have the required permission to view users that he/she has no access to. The users panel shows a list of users that do not belong to any organization. Organization admin shouldn't see the users panel on the 'Browse' tab since he/she doesn't have permission to access users outside of his/her organization.

I reverted Istvan's change and added a check for the users panel so that users search results can be displayed. Please let me know if you have any questions. Thanks. :)

pei-jung commented:
Pull request submitted to https://github.com/drewbrokke/liferay-portal/pull/78.

lipusz commented:
Hi,


"Organization admin shouldn't see the users panel on the 'Browse' tab since
he/she doesn't have permission to access users outside of his/her
organization." Actually, it's not that simple. If user has VIEW permission
on a user, he/she should be able to see and this permission check is
ensured either via the BaseSearchResultPermFilter (index-based search) or
inline permission-check (DB-search), so I'm not convinced we really need
that condition.

Regards,
Tibor

On Wed, Aug 26, 2015 at 12:46 AM, Pei-Jung Lan <notifications@github.com>
wrote:

> Hi Tibor,
>
> Nice to meet you on github too! :P
>
> We can't remove those permission constraints because those are needed to
> hide the users panel from user who do not have the required permission to
> view users that he/she has no access to. The users panel shows a list of
> users that do not belong to any organization. Organization admin shouldn't
> see the users panel on the 'Browse' tab since he/she doesn't have
> permission to access users outside of his/her organization.
>
> I reverted Istvan's change and added a check for the users panel so that
> users search results can be displayed. Please let me know if you have any
> questions. Thanks. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pei-jung/liferay-portal/pull/57#issuecomment-134762964>
> .
>



-- 

*Tibor LIPUSZ*

Senior Software Engineer @ TS
Security SME, Subscription Services & Release, Global


*Liferay Hungary Kft.* | H-1138 Budapest, Madarász Viktor u. 47. a-b.,
Hungary

*Visit Us: **www.liferay.com <http://www.liferay.com/>**  |  Like Us:
**facebook.com/liferay
<http://facebook.com/liferay>**  |  Follow Us: twitter.com/liferay
<http://twitter.com/liferay>*


* <http://twitter.com/liferay>*

<http://www.liferay.com>
